scriptworker/test/test_worker.py

```diff
@@ -60,27 +61,32 @@ def test_main(mocker, context, event_loop):
 def test_main_sigterm(mocker, context, event_loop):
     """Test that sending SIGTERM causes the main loop to stop after the next
     call to async_main."""
-    config = dict(context.config)
-    config['poll_interval'] = 1
-    creds = {'fake_creds': True}
-    config['credentials'] = deepcopy(creds)
+    run_tasks_cancelled = event_loop.create_future()
 
-    async def async_main(*args):
+    class MockRunTasks:
+        @staticmethod
+        def cancel():
+            nonlocal run_tasks_cancelled
+            run_tasks_cancelled.set_result(True)
+
+    async def async_main(main_context, _):
+        main_context.running_tasks = MockRunTasks()
         # Send SIGTERM to ourselves so that we stop
         os.kill(os.getpid(), signal.SIGTERM)
-        return True
 
     try:
         _, tmp = tempfile.mkstemp()
         with open(tmp, "w") as fh:
-            json.dump(config, fh)
-        del(config['credentials'])
+            json.dump(context.config, fh)
```

Comment:
do you mean `context` or `main_context` here?

Reply:
I'm pretty sure `context` is what I want here - `main_context` is a bit of a workaround since we need to write everything to a file, so that `worker.main` can read that file into a new `context` variable. I needed to slap `running_tasks = MockRunTasks()` onto it, and I can't encode `MockRunTasks()` into a file, so I couldn't put it on beforehand. The solution was to:
* Differentiate context before being written to file (`context`) and context after being read from the file (`main_context`)
* Modify the instance of context used in `_handle_sigterm` to that it would trigger `cancel()`

Reply:
Ok. This isn't clear during a casual read of the test. Could you add a comment, maybe similar to the above, to the test?